predicators/envs/doors.py

```diff
@@ -729,3 +738,354 @@ def _get_open_door_target_value(mass: float, friction: float,
         # A made up complicated function.
         return np.tanh(target_rot) * (np.sin(mass) +
                                       np.cos(friction) * np.sqrt(mass))
+
+
+class DoorKnobsEnv(DoorsEnv):
+    """A 2D navigation environment with obstacles, rooms, and doors."""
+    # Types
+    _robot_type = Type("robot", ["x", "y"])
```

Comment:
i think you can just inherit the `robot`, `room` and `obstacle` types from the superclass?

Reply:
yes! sorry I changed this but I guess my changes didn't save :,)